### PR TITLE
Recognize `=` as equivalent to `?` in JSDoc signatures

### LINKED ISSUE #6811: JsDoc inline function signature does not honor optional params
The below two should be equivalent, however the optionality is lost of the second param on f6.

```javascript
/** @type {function(string, boolean=): number} */
var f6;

/**
 * @param {string} p1
 * @param {boolean=} p2
 * @return {number}
 */
function f7(p1,p2){}
```

### PR DESCRIPTION
**Pull request by RyanCavanaugh**
Fixes #6811

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -5727,6 +5727,9 @@ namespace ts {
             function parseJSDocParameter(): ParameterDeclaration {
                 const parameter = <ParameterDeclaration>createNode(SyntaxKind.Parameter);
                 parameter.type = parseJSDocType();
+                if (parseOptional(SyntaxKind.EqualsToken)) {
+                    parameter.questionToken = createNode(SyntaxKind.QuestionToken);
```
Comment (DanielRosenwasser) [inline src/compiler/parser.ts]: Shouldn't this be what we consider a "synthetic" node? That usually uses an `start` of `-1` or something.

Comment (RyanCavanaugh) [inline src/compiler/parser.ts]: It's not synthetic -- it's backed by a real span

Comment (billti) [inline src/compiler/parser.ts]: Shouldn't the SyntaxKind match the actual contents of token? Here we are marking a '=' token as a QuestionToken.

I get it works, but seems a little *dirty*. Is there precedence in the code for marking tokens with a SyntaxKind which doesn't match the token contents? If not, doesn't seem like we should start.

**Comment by RyanCavanaugh** [comment]
Any other comments? @billti ?

**Comment by billti** [comment]
:+1:

**Pull request merged**